internal/testcontainersdocker/docker_host.go

```diff
@@ -29,33 +51,184 @@ func DefaultGatewayIP() (string, error) {
 	return ip, nil
 }
 
-// Extracts the docker host from the context, or returns the default value
-func ExtractDockerHost(ctx context.Context) (dockerHostPath string) {
-	if dockerHostPath = os.Getenv("TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE"); dockerHostPath != "" {
-		return dockerHostPath
+// ExtractDockerHost Extracts the docker host from the different alternatives, caching the result to avoid unnecessary
+// calculations. Use this function to get the actual Docker host. This function does not consider Windows containers at the moment.
+// The possible alternatives are:
+//
+//  1. Docker host from the "testcontainers.host" property in the ~/.testcontainers.properties file.
+//  2. DOCKER_HOST environment variable.
+//  3. Docker host from context.
+//  4. Docker host from the default docker socket path, without the unix schema.
+//  5. Docker host from the "docker.host" property in the ~/.testcontainers.properties file.
+//  6. Rootless docker socket path.
+//  7. Else, an empty string is returned.
+func ExtractDockerHost(ctx context.Context) string {
+	dockerHostOnce.Do(func() {
+		dockerHostCache = extractDockerHost(ctx)
+	})
+
+	return dockerHostCache
+}
+
+// ExtractDockerSocket Extracts the docker socket from the different alternatives, removing the socket schema and
+// caching the result to avoid unnecessary calculations. Use this function to get the docker socket path,
+// not the host (e.g. mounting the socket in a container). This function does not consider Windows containers at the moment.
+// The possible alternatives are:
+//
+//  1. The TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE environment variable.
+//  2. Using a Docker client, check if the Info().OperativeSystem is "Docker Desktop" and return the default docker socket path for rootless docker.
+//  3. Else, Get the current Docker Host from the existing strategies: see ExtractDockerHost.
+//  4. If the socket contains the unix schema, the schema is removed (e.g. unix:///var/run/docker.sock -> /var/run/docker.sock)
+//  5. Else, the default location of the docker socket is used (/var/run/docker.sock)
+func ExtractDockerSocket(ctx context.Context) string {
+	dockerSocketPathOnce.Do(func() {
+		dockerSocketPathCache = extractDockerSocket(ctx)
+	})
+
+	return dockerSocketPathCache
+}
+
+// extractDockerHost Extracts the docker host from the different alternatives, without caching the result.
+// This internal method is handy for testing purposes.
+func extractDockerHost(ctx context.Context) string {
+	dockerHostFns := []func(context.Context) (string, error){
+		testcontainersHostFromProperties,
```

Comment:
@cristianrgreco @kiview I'm implementing the new strategy here. See https://github.com/testcontainers/testcontainers-go/pull/1161/files#diff-6063b4272c0c8023987dd0a86e67194c26ff5d12a21e45ef0c90685e0b35eff0R219